3-extra/2-array-of-objects.js

```diff
@@ -11,7 +11,22 @@
 */
 
 function getHighestRatedInEachGenre(books) {
-    // TODO
+     const result = books.reduce((acc, cur) => {
```

Comment:
I think this could be simplified.You did a really good job with : 
```
      if (!acc[groupByGenre]) {
        acc[groupByGenre] = [];
      }
```
basically your accumulator is an empty object in the beginning, and you are registering the items grouping them by genre. But we may save some time if, instead of saving an array and then sorting it, we save only the highest value.
**_How do we know if a value is the highest?_**
Well, we don't know in the beginning, so we can push it the first rating in the first loop, and then on each cycle, we check if that value is higher than our previous value, and if so we override it!
```
if (cur.rating > acc[groupByGenre]){

acc[groupByGenre] = cur.rating
}
```
then our acc will look something like: acc={children: 10, fiction: 8, cooking: 9}